File: back-end/src/main/java/server/model/StudentskaSluzba.java
package server.model;

import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class StudentskaSluzba {

	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "studentskaSluzba")
	private List<Objava> objave;

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "studentskaSluzba")
	private List<Inventar> inventari;

	@OneToOne(fetch = FetchType.LAZY, mappedBy = "studentskaSluzba", optional = true, cascade = CascadeType.ALL, orphanRemoval = true)
    private Biblioteka biblioteke;



	@OneToMany(fetch = FetchType.LAZY, mappedBy = "studentskaSluzba")
	private List<Kalendar> kalendari;

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "studentskaSluzba")
	private List<Student> studenti;
	
	@OneToMany(fetch = FetchType.LAZY, mappedBy = "studentskaSluzba")
	private List<SvObrazac> obrasci;

	@Column(nullable = false)
    private Boolean vidljiv = true;
	
    public StudentskaSluzba(Long id, Boolean vidljiv) {
        this.id = id;
        this.vidljiv = vidljiv;
    }
}

File: front-end/src/app/models/StudentskaSluzba.ts
import { Objava } from './Objava';
import { Inventar } from './Inventar';
import { Biblioteka } from './Biblioteka';
import { Osoblje } from './Osoblje';
import { Nastavnik } from './Nastavnik';
import { Kalendar } from './Kalendar';
import { Student } from './Student';
import { SvObrazac } from './SvObrazac';

export interface StudentskaSluzba {
  naziv: string;
  id: number;
  vidljiv: boolean;
  objave: Objava[];
  inventari?: Inventar[];
  biblioteka: Biblioteka;
  // osoblje?: Osoblje[];
  kalendari?: Kalendar[];
  studenti?: Student[];
  obrasci?: SvObrazac[];
}
